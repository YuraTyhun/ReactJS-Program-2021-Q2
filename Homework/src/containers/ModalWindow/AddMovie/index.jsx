import React from 'react';
import { Formik, Form } from 'formik';
import PropTypes from 'prop-types';
import { useSelector } from 'react-redux';

import MovieForm from '../../MovieForm';

const AddMovie = ({ modalTitle }) => {
    const activeModal = useSelector((state) => state.movie.activeModal);

    const initialValues = {
        title: '',
        releaseDate: '',
        posterUrl: '',
        genres: [],
        overview: '',
        runtime: null
    };

    const submitForm = (values) => {
        console.log(values);
    }

    return activeModal === 'add' && (
        <Formik initialValues={initialValues} onSubmit={submitForm}>
            {(formik) => (
                <Form>
                    <h1 className="modal-window-title">{modalTitle}</h1>
                    <MovieForm formik={formik} />
                </Form>
            )}
        </Formik>
    )
}

AddMovie.propTypes = {
    modalTitle: PropTypes.string.isRequired,
};

export default AddMovie;