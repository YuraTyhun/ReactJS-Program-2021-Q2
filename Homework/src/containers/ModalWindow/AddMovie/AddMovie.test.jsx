import React from 'react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import AddMovie from './index';

it('renders addMovie form correctly', () => {
    const mockStore = configureMockStore();
    const initState = {
        movie: {
            activeModal: 'add'
        }
    };
    
    const store = mockStore(initState);
    const addMovie = renderer.create(
        <Provider store={store}>
            <AddMovie modalTitle="ADD MOVIE"/>
        </Provider>
    ).toJSON();

    expect(addMovie).toMatchSnapshot();

});