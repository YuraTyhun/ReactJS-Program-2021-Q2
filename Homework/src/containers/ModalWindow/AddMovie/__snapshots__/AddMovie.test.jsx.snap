// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders addMovie form correctly 1`] = `
<form
  action="#"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <h1
    className="modal-window-title"
  >
    ADD MOVIE
  </h1>
  <div
    className="form-input-container"
  >
    <label
      className="form-input-label"
      htmlFor="title"
    >
      TITLE
    </label>
    <input
      className="form-input-field"
      name="title"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Movie title here"
      type="text"
      value=""
    />
    
  </div>
  <div
    className="form-input-container"
  >
    <label
      className="form-input-label"
      htmlFor="release_date"
    >
      RELEASE DATE
    </label>
    <input
      className="form-input-field"
      name="release_date"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Select Date"
      type="date"
      value=""
    />
    
  </div>
  <div
    className="form-input-container"
  >
    <label
      className="form-input-label"
      htmlFor="poster_path"
    >
      MOVIE URL
    </label>
    <input
      className="form-input-field"
      name="poster_path"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Movie URL here"
      type="text"
      value=""
    />
    
  </div>
  <div
    className="form-input-container"
  >
    <label
      className="form-input-label"
      htmlFor="genres"
    >
      GENRE
    </label>
    <div
      aria-expanded={false}
      aria-owns={null}
      className="MuiAutocomplete-root form-input-field MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
      name="genres"
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="combobox"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
          onClick={[Function]}
        >
          <input
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-controls={null}
            aria-invalid={false}
            autoCapitalize="none"
            autoComplete="off"
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
            disabled={false}
            id="genres"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
            placeholder="Select genre"
            required={false}
            spellCheck="false"
            type="text"
            value=""
          />
          <div
            className="MuiAutocomplete-endAdornment"
          >
            <button
              aria-label="Clear"
              className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              title="Clear"
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
              </span>
            </button>
            <button
              aria-label="Open"
              className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              title="Open"
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </span>
            </button>
          </div>
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
            style={
              Object {
                "paddingLeft": 8,
              }
            }
          >
            <legend
              className="PrivateNotchedOutline-legend-2"
              style={
                Object {
                  "width": 0.01,
                }
              }
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <span
      className="form-input-error"
    />
  </div>
  <div
    className="form-input-container"
  >
    <label
      className="form-input-label"
      htmlFor="overview"
    >
      OVERVIEW
    </label>
    <input
      className="form-input-field"
      name="overview"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Overview here"
      type="text"
      value=""
    />
    
  </div>
  <div
    className="form-input-container"
  >
    <label
      className="form-input-label"
      htmlFor="runtime"
    >
      RUNTIME
    </label>
    <input
      className="form-input-field"
      name="runtime"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Runtime here"
      type="text"
      value={0}
    />
    
  </div>
  <div
    className="form-input-container"
  >
    <label
      className="form-input-label"
      htmlFor="budget"
    >
      BUDGET
    </label>
    <input
      className="form-input-field"
      name="budget"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Budget here"
      type="text"
      value={0}
    />
    
  </div>
  <div
    className="form-input-container"
  >
    <label
      className="form-input-label"
      htmlFor="revenue"
    >
      REVENUE
    </label>
    <input
      className="form-input-field"
      name="revenue"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Revenue here"
      type="text"
      value={0}
    />
    
  </div>
  <div
    className="form-input-container"
  >
    <label
      className="form-input-label"
      htmlFor="tagline"
    >
      TAGLINE
    </label>
    <input
      className="form-input-field"
      name="tagline"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Tagline here"
      type="text"
      value=""
    />
    
  </div>
  <div
    className="form-input-container"
  >
    <label
      className="form-input-label"
      htmlFor="vote_average"
    >
      VOTE AVERAGE
    </label>
    <input
      className="form-input-field"
      name="vote_average"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Vote average here"
      type="text"
      value={0}
    />
    
  </div>
  <div
    className="form-input-container"
  >
    <label
      className="form-input-label"
      htmlFor="vote_count"
    >
      VOTE COUNT
    </label>
    <input
      className="form-input-field"
      name="vote_count"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Vote count here"
      type="text"
      value={0}
    />
    
  </div>
  <div
    className="modal-window-action"
  >
    <button
      className="modal-window-action-btn"
      disabled={false}
      onClick={[Function]}
      type="reset"
    >
      RESET
      
    </button>
    <button
      className="modal-window-action-btn"
      disabled={false}
      onClick={[Function]}
      type="submit"
    >
      SAVE
      
    </button>
  </div>
</form>
`;
